### DISCUSSION TITLE
ETL data from clickhouse to oracle using IBM Datastage

### DISCUSSION THREAD

**Discussion by khangle97** [ClickHouse/ClickHouse#64350] [Q&A]
In IBM DataStage, I used a JDBC connector for ClickHouse and encountered this error :
JDBC_Connector_11: The connector encountered a Java exception: 
java.lang.Throwable
	at com.ibm.is.cc.jdbc.CC_JDBCRecordDataSetProducer.mapJDBCColumnsToDAAPIDataFields(CC_JDBCRecordDataSetProducer.java:1271)
	at com.ibm.is.cc.jdbc.CC_JDBCRecordDataSetProducer.setupOperation(CC_JDBCRecordDataSetProducer.java:917)
	at com.ibm.is.cc.jdbc.CC_JDBCAdapter.preRun(CC_JDBCAdapter.java:665)
	
main_program: Fatal Error: The program variable _jdbcResultSetMetaData has a null value, which is not valid in the current context. 

![image](https://github.com/ClickHouse/ClickHouse/assets/51714043/694de951-ef4b-40ce-9e88-af7bb33a4c30)

I used ClickHouse JDBC Driver 0.6.0 for job
plz help !!!

**Comment by den-crane** [comment]
1. the error is inside com.ibm.is.cc.jdbc nothing points to Clickhouse or to clickhouse-jdbc driver
2. JDBC repo is here https://github.com/ClickHouse/clickhouse-java

**Reply by khangle97** [reply]
![image](https://github.com/ClickHouse/ClickHouse/assets/51714043/5adfe1af-6b95-45d1-8c95-87651a14170c)
![image](https://github.com/ClickHouse/ClickHouse/assets/51714043/0c2d0cae-7437-4c75-b823-cb26f9773c2d)
it has used this library

**Reply by khangle97** [reply]
@den-crane
